### PR TITLE
fix(sdk): Support returning plain tuples in lightweight function components v2.

### PR DESCRIPTION
**Pull request by neuromage**
Fixes a bug where lightweight functions v2 did not support returning plain tuples.

- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 
- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

### CODE REVIEW

File: sdk/python/kfp/components/executor_test.py [hunk 1]
```diff
@@ -412,31 +413,40 @@ def test_func() -> NamedTuple('Outputs', [
...
-              'r') as f:
-      artifact_payload = f.read()
-    self.assertEqual(artifact_payload, "Dataset contents")
+    # Functions returning plain tuples should work too.
+    def func_returning_plain_tuple() -> NamedTuple('Outputs', [
+        ("output_dataset", Dataset),
+        ("output_int", int),
+        ("output_string", str),
+    ]):
+      return ("Dataset contents", 101, "Some output string")
```
Comment (chensun) [inline sdk/python/kfp/components/executor_test.py:422]: Interesting. I would have thought this as a user code error for not returning an object of the declared return type (understand Python doesn't block this).
Feels to me what you did in this PR is more like a feature to add tolerance to the system than fixing a bug.

**Review by chensun — COMMENTED** [review]
/lgtm 

Thanks!

File: sdk/python/kfp/components/executor_test.py (hunk 1 above)
Comment (neuromage) [reply sdk/python/kfp/components/executor_test.py:422]: Yeah exactly :-) In some sense, we need the named tuple so we can figure out the outputs and name them.

Comment (Ark-kun) [reply sdk/python/kfp/components/executor_test.py:422]: I deliberated about this quite a lot.
On one hand, returning a simple tuple or a list is not exactly correct.

On the other hand, re-declaring the NamedTuple is too much boilerplate.

Simple tuples are used a lot as Python return values, so I've decided to enable this style as well (the declaration still needs NamedTuple to give names and types to the outputs).

**Pull request merged**